haystack/__init__.py

```diff
@@ -1,27 +1,83 @@
-# pylint: disable=wrong-import-position,wrong-import-order
-
-from typing import Union
-from types import ModuleType
+# pylint: disable=wrong-import-position
+# Logging is not configured here on purpose, see https://github.com/deepset-ai/haystack/issues/2485
 
-try:
-    from importlib import metadata
-except (ModuleNotFoundError, ImportError):
-    # Python <= 3.7
-    import importlib_metadata as metadata  # type: ignore
+from importlib import metadata
 
 __version__: str = str(metadata.version("farm-haystack"))
 
+from generalimport import generalimport
 
-# Logging is not configured here on purpose, see https://github.com/deepset-ai/haystack/issues/2485
-import logging
-
-import pandas as pd
+generalimport(
+    # "pydantic",  # necessary for dataclasses
+    # "pandas",  # necessary for dataclasses
+    # "requests", # Used in dcdocumentstore, probably can be removed with some work
```

Comment:
IMO this is ok to have as a base dependency, I'm thinking about nodes using OpenAI that as a user I'd like to use with just `pip install farm-haystack`